packages/ap-storage/docs/async_uploads.rst

```diff
@@ -0,0 +1,229 @@
+You can register the URLs by calling :meth:`~alliance_platform.storage.registry.AsyncFieldRegistry.get_url_patterns`.
+
+.. code-block:: python
+
+       from alliance_platform.storage.registry import default_async_field_registry
+
+       urlpatterns = [
+           # ... other patterns ...
+           path("async-uploads/", include(default_async_field_registry.get_url_patterns())),
+       ]
+
+.. note::
+
+    If you use multiple registries, you will need to do this for each registry. In most cases the default registry
+    is sufficient.
+
+Cleanup command
+^^^^^^^^^^^^^^^
+
+Intermediate files are stored in the :class:`alliance_platform.storage.models.AsyncTempFile` table. Periodically clean up these files by running
+the :djmanage:`cleanup_async_temp_files` command:
+
+.. code-block:: bash
+
+    python manage.py cleanup_async_temp_files
+
+
+How it works
+------------
+
+The AsyncFile feature works in conjunction with :class:`~alliance_platform.storage.views.GenerateUploadUrlView`. The view generates a URL (e.g., a signed URL when using S3) that the frontend can then use for direct uploads. Each view is tied to a specific registry, which you can specify using ``async_field_registry`` (defaults to :data:`~alliance_platform.storage.registry.default_async_field_registry`).
+In most cases, a single registry is fine and you don't need to explicitly reference it. 
+
+The flow for async file uploads is as follows:
+
+1. When a form is rendered on the frontend (e.g., using :class:`~alliance_platform.storage.fields.async_file.AsyncFileFormField`), it knows the ``async_field_id`` from the registry and the ``generate_upload_url`` endpoint.
+
+2. When an upload occurs, the frontend first hits the ``generate_upload_url`` endpoint, passing the ``async_field_id``, filename, and optionally an ``instance_id`` for updates.
+
+3. :class:`~alliance_platform.storage.views.GenerateUploadUrlView` looks up the registry for the ``async_field_id``, checks permissions, and creates an :class:`~alliance_platform.storage.models.AsyncTempFile` record.
+
+4. The frontend receives the upload URL and uploads the file directly to the storage backend. The key for the :class:`~alliance_platform.storage.models.AsyncTempFile` is stored in the form for submission.
+
+5. Upon form submission, the backend moves the file from its temporary location to its final destination, and cleans up the :class:`~alliance_platform.storage.models.AsyncTempFile` record.
+
+6. If form submission never occurs, for example the user abandons the form after uploading a file, then the file will be retained until
+   the :djmanage:`cleanup_async_temp_files` command is run.
+
+Usage
+-----
+
+1. Add a :class:`~alliance_platform.storage.fields.async_file.AsyncFileField`: or :class:`~alliance_platform.storage.fields.async_file.AsyncImageField` to a model, optionally passing
+   the ``storage`` option depending on what the :setting:`STORAGES <django:STORAGES>`
```

Comment:
Maybe change this line to  something like

the ``storage`` option if you need to use a different backend from the project :setting:`STORAGE <django:STORAGES>` setting

Reply:
thanks, good one - fixed